examples/image_processing/image_view.py

```diff
@@ -7,43 +7,36 @@
 from marker import ImageMarker
 from pyqtgraph import mkPen
 
+
 class ImageViewer(Display):
     def __init__(self, parent=None, args=None):
         super(ImageViewer, self).__init__(parent=parent, args=args)
-        self.image_channel = "ca://MTEST:TwoSpotImage"
-        self.image_width_channel = "ca://MTEST:ImageWidth"
-        self.ui.imageView.widthChannel = self.image_width_channel
-        self.ui.imageView.image_channel = ""
+        self.ui.imageView.process_image = self.process_image
         self.markers = []
-        
+
     def ui_filename(self):
         return 'image_view.ui'
-        
+
     def ui_filepath(self):
         return path.join(path.dirname(path.realpath(__file__)), self.ui_filename())
-    
-    @pyqtSlot(np.ndarray)
-    def new_image_received(self, new_waveform):
-        #Reshape the 1D waveform into 2D
-        img = new_waveform.reshape((int(512),-1), order='C')
-        #Find blobs in the image with scikit-image
-        blobs= blob_doh(img, max_sigma=512, min_sigma=64, threshold=.02)
-        #Remove any existing blob markers        
+
+    def process_image(self, new_image):
```

Comment:
@hmbui the reshape is inside the ImageView widget... it is more natural for users to think about the processing of a 2D image instead of a flat 1D array... so we only call process_image after reshaping.

Reply:
👍